diff -Naur a/arch/arm/boot/dts/imx6dl-hummingboard.dts b/arch/arm/boot/dts/imx6dl-hummingboard.dts
--- a/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-03-22 18:54:01.000000000 +0100
+++ b/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-03-22 19:27:28.561585329 +0100
@@ -58,13 +58,13 @@
 			regulator-max-microvolt = <5000000>;
 		};
 	};
-
+/*
 	codec: spdif-transmitter {
 		compatible = "linux,spdif-dit";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_hummingboard_spdif>;
 	};
-
+*/
 	imx-drm {
 		compatible = "fsl,imx-drm";
 		crtcs = <&ipu1 0>, <&ipu1 1>;
@@ -74,9 +74,10 @@
 	sound-spdif {
 		compatible = "fsl,imx-audio-spdif";
 		model = "imx-spdif";
-		/* IMX6 doesn't implement this yet */
 		spdif-controller = <&spdif>;
 		spdif-out;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_hummingboard_spdif>;
 	};
 
         sound-hdmi {
diff -Naur a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi
--- a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-03-22 18:54:01.000000000 +0100
+++ b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-03-22 18:28:38.000000000 +0100
@@ -63,13 +63,13 @@
 			regulator-max-microvolt = <5000000>;
 		};
 	};
-
+/*
 	codec: spdif-transmitter {
 		compatible = "linux,spdif-dit";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_cubox_i_spdif>;
 	};
-
+*/
 	imx-drm {
 		compatible = "fsl,imx-drm";
 		crtcs = <&ipu1 0>, <&ipu1 1>;
@@ -81,6 +81,8 @@
 		model = "imx-spdif";
 		spdif-controller = <&spdif>;
 		spdif-out;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_cubox_i_spdif>;
 	};
 
         sound-hdmi {
--- a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-03-20 19:08:39.169319209 +0100
+++ b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-03-20 19:08:39.185319208 +0100
@@ -144,7 +144,7 @@
 };
 
 &hdmi_core {
-        ipu_id = <1>;
+        ipu_id = <0>;
         disp_id = <0>;
         status = "okay";
 };
--- a/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-02-28 09:35:11.000000000 +0100
+++ b/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-03-17 07:01:02.258818227 +0100
@@ -127,8 +127,6 @@
 };
 
 &hdmi {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hummingboard_hdmi>;
 	ddc = <&i2c2>;
 	status = "okay";
 	crtcs = <&ipu1 0>;
@@ -138,14 +136,8 @@
         status = "okay";
 };
 
-&hdmi_cec {
-        pinctrl-names = "default";
-        pinctrl-0 = <&pinctrl_hummingboard_hdmi>;
-        status = "okay";
-};
-
 &hdmi_core {
-        ipu_id = <1>;
+        ipu_id = <0>;
         disp_id = <0>;
         status = "okay";
 };
@@ -178,7 +170,7 @@
 	clock-frequency = <100000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c2_2>;
-	status = "okay";
+	status = "disable";
 };
 
 &iomuxc {
@@ -196,12 +188,6 @@
 			>;
 		};
 
-		pinctrl_hummingboard_hdmi: hummingboard-hdmi {
-			fsl,pins = <
-				MX6QDL_PAD_KEY_ROW2__HDMI_TX_CEC_LINE 0x1f8b0
-			>;
-		};
-
 		pinctrl_hummingboard_spdif: hummingboard-spdif {
 			fsl,pins = <MX6QDL_PAD_GPIO_17__SPDIF_OUT 0x1b0b0>;
 		};
